nbm-maven-plugin/src/main/java/org/apache/netbeans/nbm/CreateClusterAppMojo.java

```diff
@@ -1091,6 +1086,12 @@ static Map<String, Set<String>> computeClusterOrdering(Map<String, Set<String>>
         return cluster2depClusters;
     }
 
+    private static Set<String> intersection(Set<String> first, Collection<String> second) {
+        HashSet<String> intersection = new HashSet<>(first);
```

Comment:
Are you 100% sure that none of the sets passed in here are ordered?  Replaced code states it maintains iteration order of `first`.

Reply:
not sure I understand the concern. The set is only used in the form of `intersection.isEmpty()`.

Maybe I should change the method to `doesIntersect()` and the impl to `return new HashSet(first).retainAll(second)`?

Reply:
Sorry for the noise. I checked against the original spec without paying enough attention to how the result was actually being used!

Reply:
actually, let me change the code a bit. Some of the if checks can be changed to `col.contains(other)`